controllers/toolchainclusterresources/toolchaincluster_resources_controller_test.go

```diff
@@ -0,0 +1,124 @@
+package toolchainclusterresources
+
+import (
+	"context"
+	"embed"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/template"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+//go:embed testdata/cluster-role.yaml
+var clusterRoleFS embed.FS
+
+//go:embed testdata/service-account.yaml
+var serviceAccountFS embed.FS
+
+func TestToolchainClusterResources(t *testing.T) {
+	// given
+	// we assume there is already a service account generated in the member operator namespaces
+	sa := &v1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ServiceAccount",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "existing-sa",
+			Namespace: test.MemberOperatorNs,
+		},
+	}
+	cl := test.NewFakeClient(t, sa)
+
+	t.Run("controller should create service account resource", func(t *testing.T) {
+		// given
+		controller, req := prepareReconcile(sa, cl, &serviceAccountFS)
+
+		// when
+		_, err := controller.Reconcile(context.TODO(), req)
+		
+		// then
+		require.NoError(t, err)
+		checkExpectedServiceAccountResources(t, cl)
+	})
+
+	t.Run("controller should create cluster role resource", func(t *testing.T) {
+		// then
```

Comment:
Also here and in the next tests the comments seems misplaced

Reply:
I saw now that there was a //given in the parent test, that's why was missing in the subtests, but the //then and //when was inverted indeed. 

Anyways I think it's fine to have //given twice.

Done https://github.com/codeready-toolchain/toolchain-common/pull/373/commits/6f6a0c01bb0bcd7fbcaf173a879d576940bd0cad